packages/venia-ui/lib/components/Navigation/navHeader.js

```diff
@@ -26,11 +26,17 @@ const NavHeader = props => {
     const title = titles[view] || titles.MENU;
     const backIcon = isTopLevel && view === 'MENU' ? MenuIcon : ArrowLeftIcon;
 
+    const backButton = !isTopLevel ? (
+        <Trigger key="backButton" action={onBack}>
+            <Icon src={backIcon} />
+        </Trigger>
+    ) : (
+        <div />
```

Comment:
Do we need to render an empty `div`? Is this for spacing?

Reply:
Yes it is to adjust spacing. I was unable to find any other way.
If you have any other suggestion, please let me know.

Reply:
@sirugh : Please help to have a look at the changes, I have worked on the concerns shared by you.

Reply:
> Yes it is to adjust spacing. I was unable to find any other way. If you have any suggestion regarding this, please let me know.

That's certainly one way to do it. 😅

Since you're already making the grid layout dynamic based on whether it's the root, you could just change the number of columns and add some left padding. Then you could render `null` instead of an empty `div`.

```css
.isRoot {
  grid-template-columns: 1fr 3.5rem; /* title, close icon */
  padding-left: 1rem;  /* so the title isn't up against the edge */
}
```

Also, for what it's worth, we don't do class concatenation in our components. Each element should have no more than one classname from the imported object; you can use `composes` to combine styles.

```jsx
const Foo = props => {
  const classes = mergeClasses(defaultClasses, props.classes)

  // `.root` might have some styles of its own
  // `.root_open` and `.root_closed` might compose `.root`
  // and add some extra styles
  const rootClass = props.isOpen ? classes.root_open : classes.root_closed

  return (
    <div className={rootClass} />
  )
}
```

Reply:
@jimbo : Thanks for those suggestions, I have worked on the same. Please review.
Thanks.